controllers/utils/validation.go

```diff
@@ -0,0 +1,24 @@
+package utils
+
+import (
+	"fmt"
+
+	"github.com/medik8s/node-healthcheck-operator/api/v1alpha1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+// ValidateMinHealthy validates the minHealthy field
+func ValidateMinHealthy(nhc *v1alpha1.NodeHealthCheck) error {
+	if nhc.Spec.MinHealthy == nil {
+		return fmt.Errorf("MinHealthy is nil")
+	}
+	if nhc.Spec.MinHealthy.Type == intstr.Int && nhc.Spec.MinHealthy.IntVal < 0 {
```

Comment:
IIUC MinHealthy can be both a string and an int.
Why not just use one type (i.e do we have a justification for each use case ?)

Reply:
int is the natural fit for absolute number of nodes, and we need string for percentage. That´s exacty what IntOrString is for 🙂

Reply:
Cool, wasn't aware of the Percentage option :+1: